src/client/app/index.ts

```diff
@@ -119,21 +119,14 @@ function newApp(): App {
 
 function newRouter(): Router {
   let isInitialPageLoad = inBrowser
-  let initialPath: string
 
   return createRouter((path) => {
     let pageFilePath = pathToFile(path)
     let pageModule = null
 
     if (pageFilePath) {
+      // use lean build if this is the initial page load
```

Comment:
Because in 3.5 the static vnodes are cached per-instance instead of globally hoisted, we can no longer rely on adopted content during hydration when navigating back to the initial page. We have to load the full chunk here in that case.